### ISSUE TITLE
Files have two icons next to them

### ISSUE THREAD

**Issue by ARMIsNOTLoaded** [FlorianWoelki/obsidian-icon-folder#198]
**Describe the bug**
Suddenly some files have two icons next to them, one is unclearable, I tried disabling the plugin, uninstalling it and/or the icon pack, moving the icon pack, nothing worked.

**To Reproduce**
I have no idea how to reproduce this bug, it appeared randomly one day.

**Expected behavior**
That the files have just one icon, the default behavior of the plugin.

**Screenshots**
![image](https://github.com/FlorianWoelki/obsidian-icon-folder/assets/138717625/d4fe4427-2b6f-4906-89f8-af8e9295d95c)

**Additional context**
Knowing how to reset everything and clear icon's data inside a vault would be a great help for now.

**UPDATE**: I found the problem.
The issue appears only on PNG files I link in notes: the PNG file gets assigned the same icon as the note, and if you add another icon to the PNG itself it will show two icons. There is no way to get rid to the note's icon unless you remove it from the note where the PNG was embedded.

IDK if this is a normal behavior, it didn't do that until the day I opened this thread. I will behave accordingly knowing that.

**UPDATE 2**: it seems to trigger between folders and files if they contain the same words.
As an example, I have a folder named "Monsters" with an icon "dog".
Then if I create, anywhere in the vault, a note with "Monsters" in its name (i.e. "Amulet of Monsters").
This note WILL have the "dog" icon regardless. Can't remove it unless I remove "Monsters" from its name or remove the icon from the "Monsters" folder. If I assign another icon to the note, it will show two icons.

**Comment by RazMag** [comment]
Not a solution but I believe you can remove `VAULT-FOLDER\.obsidian\plugins\obsidian-icon-folder\data.json` to 'reset' your vault. if you have a syncing solution this might be problematic though

**Comment by ARMIsNOTLoaded** [comment]
> Not a solution but I believe you can remove `VAULT-FOLDER\.obsidian\plugins\obsidian-icon-folder\data.json` to 'reset' your vault. if you have a syncing solution this might be problematic though

I want to thank you because this solved my problem entirely and the plugin is now working fine.

**Comment by Net-Slayer** [comment]
I think this might be the culprit? from inspecting the packaged main.js
```js
const doesExistInPath = (plugin, path) => {
    const folders = getFolders(plugin);
    return Object.keys(folders).some((folderPath) => path.includes(folderPath));
};
```

for me I experience the same issue with  "05. Meta/Templates" and "05. Meta/Temp", where the icons double up because one can be considered a substring of another.

```js
const inputString = "05. Meta/Templates";
const searchString = "05. Meta/Temp";

const result = inputString.includes(searchString);

console.log(result); // Output: true
```

**Comment by califa** [comment]
Also experiencing this, up to 3 icons so far :)

<img width="88" alt="image" src="https://github.com/FlorianWoelki/obsidian-icon-folder/assets/602352/13c99cf2-37ae-49f7-8d67-6f0226b0cc4a">

**Comment by FlorianWoelki** [comment]
Could you please give a rough outline on how to reproduce this issue with minimal effort?

Thank you!

**Comment by Net-Slayer** [comment]
Sure, my tests were with the word "Understandable"

Create the following folders/files for testing:
* Under/Stand/Able.md
* Under/Able/Stand.md
* Able/Stand/Under.md
* Able/Under/Stand.md
* Stand/Under/Able.md
* Stand/Able/Under.md

* Under/Understand.md
* Understand/Under.md

* Under/Understand/Understandable.md
* Understandable/Understand/Under.md

Assign inheritance icons to each top and mid level folder.
Assign icons to files.

You should notice stacking icons.
This is due to path.includes(), but it's ok it just needs a more sophisticated check for parent folders/child folders in inheritance.ts

**Comment by FlorianWoelki** [comment]
This should be fixed in [2.2.0](https://github.com/FlorianWoelki/obsidian-icon-folder/releases/tag/2.2.0)
